operator/builtin/input/windows/operator.go

```diff
@@ -0,0 +1,223 @@
+// +build windows
+
+package windows
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/observiq/carbon/operator"
+	"github.com/observiq/carbon/operator/helper"
+)
+
+func init() {
+	operator.Register("windows_eventlog_input", NewDefaultConfig)
+}
+
+// EventLogConfig is the configuration of a windows event log operator.
+type EventLogConfig struct {
+	helper.InputConfig `yaml:",inline"`
+	Channel            string            `json:"channel" yaml:"channel"`
+	MaxReads           int               `json:"max_reads,omitempty" yaml:"max_reads,omitempty"`
+	StartAt            string            `json:"start_at,omitempty" yaml:"start_at,omitempty"`
+	PollInterval       operator.Duration `json:"poll_interval,omitempty" yaml:"poll_interval,omitempty"`
+}
+
+// Build will build a windows event log operator.
+func (c *EventLogConfig) Build(context operator.BuildContext) (operator.Operator, error) {
+	inputOperator, err := c.InputConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Channel == "" {
+		return nil, fmt.Errorf("missing required `channel` field")
+	}
+
+	if c.MaxReads < 1 {
+		return nil, fmt.Errorf("the `max_reads` field must be greater than zero")
+	}
+
+	if c.StartAt != "end" && c.StartAt != "beginning" {
+		return nil, fmt.Errorf("the `start_at` field must be set to `beginning` or `end`")
+	}
+
+	offsets := helper.NewScopedDBPersister(context.Database, c.ID())
+
+	eventLogInput := &EventLogInput{
+		InputOperator: inputOperator,
+		buffer:        NewBuffer(),
+		channel:       c.Channel,
+		maxReads:      c.MaxReads,
+		startAt:       c.StartAt,
+		pollInterval:  c.PollInterval,
+		offsets:       offsets,
+	}
+	return eventLogInput, nil
+}
+
+// NewDefaultConfig will return an event log config with default values.
+func NewDefaultConfig() operator.Builder {
+	return &EventLogConfig{
+		MaxReads: 100,
+		StartAt:  "end",
+		PollInterval: operator.Duration{
+			Duration: 5 * time.Second,
```

Comment:
The way I understand it, the defaults here will limit us to processing 100 logs every 5 seconds. Is that correct?

Reply:
Yes. I chose that default, because in my experience, windows events does not seem to have a constant/heavy stream of traffic. It felt like overkill to poll any faster than this. I could possibly reduce the default polling to maybe 1 second, but even that seems like a lot.

Reply:
Okay. I'm hesitant on this because 20 logs a second seems like it could very quickly be overwhelmed, and we don't warn the user at all that we're not keeping up. The MO in other plugins is just "read as fast as you can, then use a rate limiter if you really care about slowing it down". Additionally, when demoing, it's really nice to get instant results, and 5 seconds a bit long to be considered "instant". 

Idea: Would it be possible to do something like the file input plugin? Behavior is more like "Every 5 seconds, poll to see if there are more logs to read. If there are, continue reading in batches until we reach the end of the event log"

Reply:
Yeah, I wouldn't be opposed to that solution. The main thing I want to avoid is what we've encountered with windows events before, where we take up a bunch of memory reading all of the events from the beginning, rather than the end. The poll interval was meant to help distribute that load. This new solution should still help prevent that, but I think we need to be careful. It would be really easy to prevent the agent from shutting down while reading a large number of logs, since the context is being checked with the ticker.

Reply:
RE: memory issues, with the batch size, am I correct that we are never be loading more than a single batch into memory at a time? If that's true, it seems like your solution will still protect against that

RE: cancellation, it should be possible to pass the context down safely as well, cancelling the "read to end" as soon as the current batch (or even current entry) is complete

Reply:
I think what I'm going to do is keep the Subscription struct as is, since it's pretty clean and matches the API. I don't want to pass any type of context to it and complicate that logic. Instead, I'll let the plugin handle that logic. It will keep calling Read on the subscription until the result set is smaller than the max reads, at which point it will return to the ticker. And after each call to read, I'll make sure the context hasn't been cancelled.